migrationtools/migrations.py

```diff
@@ -305,27 +350,121 @@ def migrate_nurseconnect_registrations(self, start=None, stop=None, limit=None):
         migration_id = str(uuid.uuid4())
         migration_type = 'nurseconnect'
         self.echo('Starting nurseconnect migration ({0})'.format(migration_id))
-        for registration in self.ndoh_control.get_registrations(start, stop, limit=limit):
+        for registration in self.ndoh_control.get_nc_registrations(start, stop, limit=limit):
             # Keep a record of all transactions started so they can all be rolled back on any error.
             transactions = {}
 
-            # id              | integer
-            # cmsisdn         | character varying(255)
-            # dmsisdn         | character varying(255)
-            # rmsisdn         | character varying(255)
-            # faccode         | character varying(100)
-            # id_type         | character varying(8)
-            # id_no           | character varying(100)
-            # passport_origin | character varying(100)
-            # dob             | date
-            # nurse_source_id | integer
-            # persal_no       | integer
-            # opted_out       | boolean
-            # optout_reason   | character varying(100)
-            # optout_count    | integer
-            # sanc_reg_no     | integer
-            # created_at      | timestamp with time zone
-            # updated_at      | timestamp with time zone
+            registration_id = registration[nc_cols.id]
+            self.echo("Migrating nurseconnect registration: {id} ...".format(id=registration_id), nl=False)
+
+            cmsisdn = registration[nc_cols.cmsisdn]  # contact
+            dmsisdn = registration[nc_cols.dmsisdn]  # device
+            created_at = registration[nc_cols.created_at]
+
+            # Setup a seed identity transaction:
+            transactions['seed_identity'] = self.seed_identity.start_transaction()
+
+            if dmsisdn is not None and cmsisdn == dmsisdn:
+                # This person was registered by a 3rdparty, check if they are already in the system.
```

Comment:
Can you explain this to me? It would seem like if the device msisdn is equal to the contact's msisdn it would be a self registration instead of a 3rd party?